### DISCUSSION TITLE
CentOS Stream: What is the impact for Centreon?

### DISCUSSION THREAD

**Discussion by quanghungb** [centreon-archive/centreon-archived#9317] [General]
Dear all, 

Red Hat just made an announcement about [CentOS Stream](https://www.redhat.com/en/blog/centos-stream-building-innovative-future-enterprise-linux). We took a look at this subject and shared our thoughts on [our blog](https://www.centreon.com/en/blog/centos-stream-about-the-recent-red-hat-announcement/).

Now, we need to hear from you: please share your thoughts, ideas, requirements, Linux distribution of choice here !

Below the list of current requests about this subject :

- #9282 
- #9289 

Kind regards,
The Centreon Team

**Comment by BenoitPoulet** [comment]
Hello,

It's a bit early to talk about the "after" of CentOS as the community and projects will quickly evolve after this announcement. 

But what about OpenSuse ? Or wait a bit, looking at Rocky Linux

**Reply by UrBnW** [reply]
> Or wait a bit, looking at Rocky Linux

Not so fond of experimenting a new OS, furthermore on production servers 😕
A monitoring server / monitoring infrastructure needs some kind of robustness / stability IMO.

**Reply by BenoitPoulet** [reply]
Yes , but did you see who started the project ? The original creator of CentOS.
We need to wait a bit to see how will be the project.

**Reply by UrBnW** [reply]
Sure, but then he needs a team, processes etc... it may take some time, and it may also take some time to have a quality product, something we don't need to "debug".
And of course the whole thing needs to be solid enough to remain throughout the years.

**Comment by sc979** [comment]
Maybe Debian is a solution to study too

**Reply by UrBnW** [reply]
As proposed in #9282, for its server orientation, its robustness and security positioning 👍
Not the very last packages' version, but IMO Centreon does not need this, and what is today in Debian10 should for sure do the trick.
Centreon should then not run backported packages or whatever, but stick with the distribution package versions, or we loose the Debian strategy.

**Reply by boisrobe** [reply]
Debian is already document. I read some good article on sugarbug.fr for centreon installation

**Reply by ypilpre** [reply]
Supporting Debian or  Ubuntu will also "officialy" open  Raspberry Pi OS  to  Centreon (Thanks to @kermith72 works on this platform)

**Comment by rlemerlus** [comment]
Open-source: OpenSuse, Debian, Ubuntu, CentOS7
Open-source with support: SLES, Ubuntu Server, RHEL

**Reply by lgcosta** [reply]
> Open-source: OpenSuse, Debian, Ubuntu, CentOS7
> Open-source with support: SLES, Ubuntu Server, RHEL

Open-source: BSD ! BSD ! BSD ! :pray: :joy: :joy: :joy:

**Reply by UrBnW** [reply]
Unfortunately (don't get me wrong ;) BSD (FreeBSD for instance) is by far less popular than Linux, would then be some kind of a pain for a product such as Centreon, IMO.
But let's keep supporting Beastie :wink:

**Reply by lgcosta** [reply]
yes ! you are absolutely right ! but it is good to remember that not every popular OS is the best (Windows is really popular, but it is not always considered the best). You may not know it, but FreeBSD is more popular than you think. If you use Netflix, you are using a Freebsd, perhaps, much of what surf's in your daily internet use, is going through a Juniper router that uses an OS based on FreeBSD, as well as your fun on the Playstation. Perhaps your permissive license is not good marketing for the name (and neither is the design intent). Another good example is OpenBSD, considered one of the most secure opensource OS out there, however, it is by far the most popular.

Anyway, it really is not my intention here that we can consider BSD as official support! I am aware that commercially we have Linux as a market leader, despite the dangers we have with licenses (maybe, really with Debian we are far from that). And maybe that's also why (and certainly history confirm this) we have such a permissive license with BSD.

**Comment by pkriko** [comment]
I'm not a huge fan of Oracle but apparently Oracle Linux and CentOS were pretty close :eyes:

**Reply by quanghungb** [reply]
Yes ! AFAIK Oracle Linux is compiled from RHEL. source code too.

**Reply by lgcosta** [reply]
> Yes ! AFAIK Oracle Linux is compiled from RHEL. source code too.

I am afraid of happening with Oracle Linux the same as Centos. Oracle licensing is a bit dangerous. (remember Mysql and StarOffice, in addition to the Sun stuff ... does anyone still remember OpenSolaris ?!)

**Comment by quanghungb** [comment]
One of the most important things is defining the "correct" list of selection critera in order to make a [comparison](https://en.wikipedia.org/wiki/Comparison_of_Linux_distributions) and the "right" choice, as there are [a lot of distributions](https://en.wikipedia.org/wiki/List_of_Linux_distributions) 

What would be yours ?

**Comment by ypilpre** [comment]
IMOO, Do an application like Centreon need to be specific to an Linux OS ?
 If the product is built as an appliance, OS is the choice of the editor, and the appliance could be delivered as an independent turn key platform as an cloud image or  an iso (as VMWare do with [Photon OS](https://vmware.github.io/photon/) for VSphere Appliance, or other editors).
If Centreon delivers the application as a portable application, Major Distributions (Suse, Redhat, Ubuntu) need to be supported or even in several containers for maximum portability on Cloud Platforms. 
I concede that maintaining multiple Linux OS compatibility and delivery forms are a great challenge for Q&A, packaging and Documentation Teams, but worth the spread of Centreon.

Several supervision & monitoring competitors made the leap to Redhat/SUSE/Ubuntu so why not Centreon ?

**Reply by quanghungb** [reply]
Yes, we would love to be able to deliver Centreon for some well-known distributions ;) 
But you are right about the "additional challenges" for the QA, packaging, documentation, maintenance & Support :).

**Comment by lgcosta** [comment]
Opportunity for me to update the code for alternative packages, including BSD :blush:

- FreeBSD: https://gitlab.labexposed.com/centreon-lab/freebsd-ports
- Alpine Linux: https://gitlab.labexposed.com/centreon-lab/alpine-packages

**Comment by teonitro** [comment]
hello,
 
Our centreon is very obsolete (CentOS 6 and Centreon 2.5). I have to make a migration to Centreon 20.10 but the choice of Operating System is difficult regarding RH announces. 
I don't want to choose CentOS7 because a migration in 3 years will be not easy with our environment. 
So using CentOS8 stream is not recommanded from Centreon because this release is in development branch. 
My point of view is to start to CentOS 8 legacy waiting a EL8 free OS in 2021 like rockyOS to migrate repositories to this new OS. 
RH provides officially scripts to migrate in RHEL8 : [https://access.redhat.com/articles/2360841] (my backup plan)

Do you think my plan make sens ?

Regards, 

Benoit

**Reply by lpinsivy** [reply]
Hi @teonitro, we updated documentation. You can choose RHEL or Oracle Linux 8.

**Comment by BenoitPoulet** [comment]
I just tried openSUSE **Leap**, close to CentOS.

Based on SLES, and Suse officially support the migration between Leap and SLES which is a good point.

Release cycle are between 36 or 48 month.

**Comment by teonitro** [comment]
thanks to update the documentation and support OL8. but we have make a choice to select CentOS 8 legacy and hope to migrate to OL8 now or to the RockyOS. We will make another choice to migrate at the end 2021 to the good OS. 

thanks for your support

**Comment by itsul** [comment]
Debian will be a good choice regarding the stability.

**Comment by UrBnW** [comment]
Debian 11 next year (2022) as per my understanding :
https://youtu.be/X6WWy7cKpy4
